CodeEdit/NavigatorSidebar/NavigatorSidebarToolbarBottom.swift

```diff
@@ -34,8 +34,16 @@ struct NavigatorSidebarToolbarBottom: View {
 
     private var addNewFileButton: some View {
         Menu {
-            Button("Add File") {}
-                .disabled(true)
+            Button("Add File") {
+                guard let folderURL = workspace.workspaceClient?.folderURL() else { return }
+                guard let root = try? workspace.workspaceClient?.getFileItem(folderURL.path) else { return }
+                root.addFile(fileName: "untitled") // TODO: use currently selected file instead of root
+            }
+            Button("Add Folder") {
+                guard let folderURL = workspace.workspaceClient?.folderURL() else { return }
+                guard let root = try? workspace.workspaceClient?.getFileItem(folderURL.path) else { return }
+                root.addFolder(folderName: "untitled") // TODO: use currently selected file instead of root
+            }
```

Comment:
Crashes on projects located on Desktop sometimes.

> Thread 1: EXC_BAD_ACCESS (code=1, address=0x3ff0000000000008)

Reply:
What do you mean "crashes on projects"? Do you mean when you click the right click menu items? Also, if you recreate the exact setup in another folder, does it work? I've always used a folder on my desktop to test it out and it worked.

Reply:
@lukepistrol did you have the same issue that wdg had with the addFile function not being updated with my changes?

Reply:
I have reopened https://github.com/CodeEditApp/CodeEdit/issues/628, so that we can make progress on merging this PR.